### PR TITLE
Improve the performance of the database dumper

### PR DESCRIPTION
**Pull request by Toflar**
Seems like with the latest changes in Doctrine DBAL a lot of internal deprecations are suppressed because it uses `getWrappedConnection()` inside `quote()` itself. Also, it seems to do in-memory caching there as well.

By reducing the calls to `quote()` and using an in-memory cache ourselves, we can cut both, the time needed to create a backup and the memory usage in half 🥳 

Bildschirmfoto 2024-02-27 um 17 03 50

### CODE REVIEW

File: core-bundle/src/Doctrine/Backup/Dumper.php [hunk 1]
```diff
@@ -32,10 +34,12 @@ public function dump(Connection $connection, CreateConfig $config): \Generator
             }
 
             throw new BackupManagerException($exception->getMessage(), 0, $exception);
+        } finally {
+            $this->quoteCache = [];
         }
     }
 
-    public function doDump(Connection $connection, CreateConfig $config): \Generator
+    private function doDump(Connection $connection, CreateConfig $config): \Generator
```
Comment (Toflar) [inline core-bundle/src/Doctrine/Backup/Dumper.php:42]: I think this was an oversight - this should definitely be private 😊

File: core-bundle/src/Doctrine/Backup/Dumper.php [hunk 2]
```diff
@@ -76,7 +80,7 @@ private function disableQueryBuffering(Connection $connection): void
     }
 
     /**
-     * @param AbstractSchemaManager<AbstractPlatform> $schemaManager
+     * @phpstan-param AbstractSchemaManager<AbstractPlatform> $schemaManager
```
Comment (leofeyer) [inline core-bundle/src/Doctrine/Backup/Dumper.php]: Why did you change this?

Comment (Toflar) [reply core-bundle/src/Doctrine/Backup/Dumper.php]: I did not, the CS fixer did.

**Review by aschempp — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

File: core-bundle/src/Doctrine/Backup/Dumper.php (hunk 2 above)
Comment (leofeyer) [reply core-bundle/src/Doctrine/Backup/Dumper.php]: There is no such rule in our fixer setup. 🤔 Anyways, I have reverted it in b846408d87dc5f782ca8b802dff149988fa59faa.

**Comment by leofeyer** [comment]
Thank you @Toflar.
